docs/contributing/components.md

```diff
@@ -0,0 +1,112 @@
+---
+sidebar_label: Components
+---
+
+```mdx-code-block
+import CodeBlock from '@theme/CodeBlock';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+```
+
+# Contributing to Components
+
+The source code for Resoto lives in the [`someengineering/resoto` repository on GitHub](https://github.com/someengineering/resoto).
+Resoto consists of multiple components, where each component is maintained as separate project.
+A change should target one of the components.
+
+- [Resoto Core](https://github.com/someengineering/resoto/tree/main/resotocore)
+- [Resoto Shell](https://github.com/someengineering/resoto/tree/main/resotoshell)
+- [Resoto Worker](https://github.com/someengineering/resoto/tree/main/resotoworker)
+- [Resoto Metrics](https://github.com/someengineering/resoto/tree/main/resotometrics)
+- [Resoto Library](https://github.com/someengineering/resoto/tree/main/resotolib)
+
+## Authoring Changes
+
+Contributions are made via [pull requests to the GitHub repository](https://github.com/someengineering/resoto/pulls).
+You will first need to [fork](https://docs.github.com/get-started/quickstart/fork-a-repo) the repository.
+
+#### Prerequisites
+
+- [Git](https://git-scm.com)
+- [Python](https://python.org) 3.8+ (3.10 is recommended)
+- [ArangoDB](https://arangodb.com) 3.8.2+
+- [GNU Compiler Collection (GCC)](https://gcc.gnu.org) (depending on the host system, Python dependencies may need to be compiled from source)
+
+#### Setting Up a Virtual Environment
+
+We recommend using a [Python virtual environment](https://docs.python.org/3/tutorial/venv.html).
+
+A script is provided to simplify the process of configuring the virtual environment:
+
+```bash
+./setup_venv.sh --dev --path .
+```
+
+Activate the virtual environment:
+
+```mdx-code-block
+<Tabs>
+  <TabItem value="linux" label="Linux/macOS">
+    <CodeBlock language="bash">source venv/bin/activate</CodeBlock>
+  </TabItem>
+  <TabItem value="windows" label="Windows">
+    <CodeBlock language="powershell">venv\Scripts\activate.bat</CodeBlock>
+  </TabItem>
+</Tabs>
+```
+
+#### Starting the Database
+
+Start ArangoDB (using `systemctl` on Linux, by clicking the application icon in macOS, etc.).
+
+**The expected default access is the built-in user account `root` with a configured empty string as password or authentication turned off.**
+This is necessary for `resotocore` to perform the database setup, and also required for tests to pass.
```

Comment:
```suggestion
Start ArangoDB (using `systemctl` on Linux, by clicking the application icon in macOS, etc.).

**By default, the built-in user account `root` does not have a password configured (i.e., the password defaults to an empty string).** This is necessary for `resotocore` to perform the database setup, and also required for tests to pass.
```
The main issue was "the expected default access," which sounds odd and is vague/unclear.  But I added what you have about the empty string!

Reply:
This sounds like this would be the default of the database.
That might or might not be the case.
It is important, that the user either already has this setup or configures the database this way:
- option 1: username: root, password: <empty string>
- option 2: turn authentication off

It does not seem very clear to me.

Reply:
The phrase, "by default" means if the user did not do any prior configuration.  From a quick search, it seems that this is in fact the default for new ArangoDB installs.

Reply:
But, we do need to address the case where users may have already configured ArangoDB.  I'll update my suggestion shortly.

Reply:
Yes. I installed arango on Fedora today - I needed to manually configure it.
I think it makes sense to say what we expect.
Would be great if you could make a proposal.

Reply:
@aquamatthias Yup, I wasn't aware that the default password could vary depending on the install method.  Updated my suggestion above!

Reply:
👍 thanks